sycl/plugins/unified_runtime/ur/adapters/cuda/device.cpp

```diff
@@ -873,6 +873,85 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
   case UR_DEVICE_INFO_MAX_COMPUTE_QUEUE_INDICES: {
     return ReturnValue(int32_t{1});
   }
+  case UR_DEVICE_INFO_BINDLESS_IMAGES_SUPPORT_EXP: {
+    // On CUDA bindless images are supported.
+    return ReturnValue(true);
+  }
+  case UR_DEVICE_INFO_BINDLESS_IMAGES_1D_USM_SUPPORT_EXP: {
+    // On CUDA 1D bindless image USM is not supported.
+    // More specifically, linear filtering is not supported.
+    return ReturnValue(false);
+  }
+  case UR_DEVICE_INFO_BINDLESS_IMAGES_2D_USM_SUPPORT_EXP: {
+    // On CUDA 2D bindless image USM is supported.
+    return ReturnValue(true);
+  }
+  case UR_DEVICE_INFO_IMAGE_PITCH_ALIGN_EXP: {
+    int32_t tex_pitch_align = 0;
+    sycl::detail::ur::assertion(
```

Comment:
I'm not sure I see the same vision you have, but if I do then I don't think the use case is big enough to warrant it.

There are many cases in this switch statement which do as we've done here, however the majority aren't as simple as ours and require extra assertions or computation beyond the CUDA call.

Would you be able to elaborate on what you'd like to see here? And would it just be for the 4 cases I have that apply or for all cases in the switch that use a CUDA call?

Reply:
Something like
```
<typename RetType>
void ReturnHelper(auto &&F) {
  RetType Var;
  sycl::detail::ur::assertion(F(Var));
  return ReturnValue(Var);
}

// use:
return ReturnHelper<int32_t>([](auto &text_pitch_align) {
  return cuDeviceGetAttribute(&tex_pitch_align,
                             CU_DEVICE_ATTRIBUTE_TEXTURE_PITCH_ALIGNMENT,
                             hDevice->get());
});
```

likely defined near the `ReturnValue` helper. Not sure if I'd want that complexity for just these four cases, but if we have the same anywhere else then it would be justified, IMO.